src/Interpreters/executeQuery.cpp

```diff
@@ -614,276 +357,40 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
-                elem.exception_code = getCurrentExceptionCode();
-                elem.exception = getCurrentExceptionMessage(false);
-
-                QueryStatus * process_list_elem = context.getProcessListElement();
-                const Settings & current_settings = context.getSettingsRef();
-
-                /// Update performance counters before logging to query_log
-                CurrentThread::finalizePerformanceCounters();
-
-                if (process_list_elem)
-                {
-                    QueryStatusInfo info = process_list_elem->getInfo(true, current_settings.log_profile_events, false);
-                    status_info_to_query_log(elem, info, ast);
-                }
-
-                if (current_settings.calculate_text_stack_trace)
-                    setExceptionStackTrace(elem);
-                logException(context, elem);
-
-                /// In case of exception we log internal queries also
-                if (log_queries && elem.type >= log_queries_min_type && Int64(elem.query_duration_ms) >= log_queries_min_query_duration_ms)
-                {
-                    if (auto query_log = context.getQueryLog())
-                        query_log->add(elem);
-                }
-
-                ProfileEvents::increment(ProfileEvents::FailedQuery);
-                if (ast->as<ASTSelectQuery>() || ast->as<ASTSelectWithUnionQuery>())
-                {
-                    ProfileEvents::increment(ProfileEvents::FailedSelectQuery);
-                }
-                else if (ast->as<ASTInsertQuery>())
-                {
-                    ProfileEvents::increment(ProfileEvents::FailedInsertQuery);
-                }
-
+                query_process->queryException();
             };
 
             res.finish_callback = std::move(finish_callback);
             res.exception_callback = std::move(exception_callback);
+        }
 
-            if (!internal && res.in)
-            {
-                WriteBufferFromOwnString msg_buf;
-                res.in->dumpTree(msg_buf);
-                LOG_DEBUG(&Poco::Logger::get("executeQuery"), "Query pipeline:\n{}", msg_buf.str());
-            }
+        if (!internal && res.in)
+        {
+            WriteBufferFromOwnString msg_buf;
+            res.in->dumpTree(msg_buf);
+            LOG_DEBUG(&Poco::Logger::get("executeQuery"), "Query pipeline:\n{}", msg_buf.str());
         }
     }
     catch (...)
     {
-        if (!internal)
-        {
-            if (query_for_logging.empty())
-                query_for_logging = prepareQueryForLogging(query, context);
-
-            onExceptionBeforeStart(query_for_logging, context, time_in_microseconds(current_time), ast);
-        }
+        /// executeQuery() may fail, hence res.query_process may not be set, do this again.
+        if (query_process)
+            res.query_process = std::move(query_process);
+        /// Checking that query_process is set, since something before it's creation may fail, i.e.:
```

Comment:
Why can't we always set `query_process`? Even if the query fails before execution, we add a log entry about this (query_log + server log). If `query_process` is 'everything related to query logs and process list', it applies to all queries.

Reply:
> Why can't we always set query_process?

This condition for internal queries, that do not need all of this (AFAICS), only logging to the log (plain text server log).

> Even if the query fails before execution, we add a log entry about this (query_log + server log).

Yes, there will be entry with exception before start

>If query_process is 'everything related to query logs and process list', it applies to all queries.

Yes, but logging to the server log done separately, because queries that are passed via executeQuery (HTTPHandler/TCPHandler) they are logged separately since they do some job between logging and creating QueryProcess